esp-wifi/src/compat/common.rs

```diff
@@ -308,6 +308,14 @@ pub fn receive_queued(queue: *mut c_void, item: *mut c_void, block_time_tick: u3
         yield_task();
     }
 }
+pub fn number_of_messages_in_queue(queue: *mut c_void) -> u32 {
+    trace!("queue_msg_waiting {:?}", queue);
+    if queue != unsafe { addr_of_mut!(REAL_WIFI_QUEUE).cast() } {
```

Comment:
Wouldn't `addr_of!` be better here?

Reply:
Not sure, tbh I just adapted this from the `receive_queued` function.

Reply:
I think it's because `queue` is `*mut` - when using just `addr_of!` we would need to make `queue` into `*const` - not sure if that would "look better" 🤷

Reply:
I guess that makes sense.

Reply:
In the C implementation, the queue is passed as a const pointer. If we were to change it, we'd have to change it for all functions. For now it probably doesn't make a big difference.

Reply:
Something we should cleanup in a subsequent PR

Reply:
I could do that later today. I'm gonna mark this as resolved for now.

Reply:
I adjusted this for `queue_msg_waiting`, since it's actually valid for all other functions.